Widgets/BackgroundView.cpp

```diff
@@ -54,27 +56,28 @@ void BackgroundView::WriteImage(QString fName, QString type) {
 
 QSize BackgroundView::sizeHint() const { return QSize(pixmap.width(), pixmap.height()); }
 
-void BackgroundView::paintEvent(QPaintEvent * /* event */) {
-  if (!model) return;
-
-  QPainter painter(this);
+void BackgroundView::Paint(QPainter &painter) {
+  if (!model) {
+    grid.show = false;
+    return;
+  }
 
-  painter.fillRect(QRectF(0, 0, pixmap.width() * zoom, pixmap.height() * zoom), ArtManager::GetTransparenyBrush());
+  painter.fillRect(QRectF(0, 0, pixmap.width(), pixmap.height()), ArtManager::GetTransparenyBrush());
 
-  painter.scale(zoom, zoom);
   bool transparent = false;
   painter.drawPixmap(0, 0, (transparent) ? transparentPixmap : pixmap);
 
-  bool gridVisible = model->data(Background::kUseAsTilesetFieldNumber).toBool();
-  int gridHorSpacing = model->data(Background::kHorizontalSpacingFieldNumber).toInt();
-  int gridVertSpacing = model->data(Background::kVerticalSpacingFieldNumber).toInt();
-  int gridHorOff = model->data(Background::kHorizontalOffsetFieldNumber).toInt();
-  int gridVertOff = model->data(Background::kVerticalOffsetFieldNumber).toInt();
-  int gridWidth = model->data(Background::kTileWidthFieldNumber).toInt();
-  int gridHeight = model->data(Background::kTileHeightFieldNumber).toInt();
-
-  if (gridVisible) {
-    paintGrid(painter, pixmap.width(), pixmap.height(), gridHorSpacing, gridVertSpacing, gridHorOff, gridVertOff,
-              gridWidth, gridHeight);
+  if (model->data(Background::kUseAsTilesetFieldNumber).toBool()) {
+    grid.show = true;
+    grid.horSpacing = model->data(Background::kHorizontalSpacingFieldNumber).toInt();
```

Comment:
In the future, it might be cool to allow binding the ScrollAreaBackground to these fields, directly. Especially if we add a UI-specific field to paths for persisting grid settings, as we've done for rooms.

Incidentally, maybe we should consider extracting those into a file we can `.gitignore`, eg, `roomname.rm.ide`.

Reply:
Ok I'll mark this with a to-do but I already do this in a few places. I wanted that issue resolved before this practice becomes too widespread so hopefully we can do something about either in or around the egm pr

Reply:
Paths already has this ui persistent option for grids

Reply:
Not exactly sure what you mean but paths do have a persistent grid. This sounds like a later task either way.